spec/models/case_court_report_context_spec.rb

```diff
@@ -13,274 +13,152 @@
-          expect(contact_type_and_dates[:dates]).to_not include(case_contact_3_date.strftime("%m/%d"))
-        end
-
-        it "contains a string with the dates of the case contacts with the contact type sorted by oldest date first" do
-          contact_type_and_dates = court_report_context[:case_contacts].find { |case_contact_type_and_dates|
-            case_contact_type_and_dates[:type] == contact_type_1.name
-          }
-
-          expect(contact_type_and_dates).to_not be_nil
-
-          dates = contact_type_and_dates[:dates]
-          case_contact_1_date_index = dates.index(case_contact_1_date.strftime("%m/%d"))
-          case_contact_2_date_index = dates.index(case_contact_2_date.strftime("%m/%d"))
-          case_contact_4_date_index = dates.index(case_contact_4_date.strftime("%m/%d"))
-
-          expect(case_contact_1_date_index).to be > case_contact_2_date_index
-          expect(case_contact_2_date_index).to be > case_contact_4_date_index
-        end
-
-        context "when there are multiple medium types" do
-          before(:each) do
-            case_contact_4.update_attribute(:medium_type, "voice-only")
-          end
-
-          describe ":dates_by_medium_type" do
-            it "contains a key for each medium type and the dates of the case contacts with the medium type the values" do
-              case_contact_object_containing_complex_dates_by_medium_type = court_report_context[:case_contacts].find { |case_contact_type_and_dates|
-                case_contact_type_and_dates[:type] == contact_type_1.name
-              }
-
-              dates_by_medium_type = case_contact_object_containing_complex_dates_by_medium_type[:dates_by_medium_type]
+  describe "#case_contacts" do
+    let(:casa_case) { create(:casa_case) }
+    let(:context) { build(:case_court_report_context, casa_case: casa_case) }
 
-              expect(dates_by_medium_type).to have_key(case_contact_1.medium_type)
-              expect(dates_by_medium_type).to have_key(case_contact_4.medium_type)
+    it "returns an empty array if there are no interviewees" do
+      expect(context.case_contacts).to eq([])
+    end
 
-              expect(dates_by_medium_type[case_contact_1.medium_type]).to include(case_contact_1_date.strftime("%m/%d"))
-              expect(dates_by_medium_type[case_contact_2.medium_type]).to include(case_contact_2_date.strftime("%m/%d"))
-              expect(dates_by_medium_type[case_contact_4.medium_type]).to include(case_contact_4_date.strftime("%m/%d"))
-            end
-          end
-        end
-      end
+    # not sure how to test this without just restating the code
+    # context 'when there are interviewees' do
+    #   it 'it calls CaseContactsContactDates with filtered values' do
+    #     create_list(:case_contact_contact_type, 3, case_contact: create(:case_contact, casa_case: casa_case))
+    #     context.case_contacts
+    #   end
+    # end
```

Comment:
Not sure how best to test the case where there are interviewees, should I even test it?

```rb
  def case_contacts
    cccts = CaseContactContactType.includes(:case_contact, :contact_type).where("case_contacts.casa_case_id": @casa_case.id)
    interviewees = filter_out_old_case_contacts(cccts)
    return [] unless interviewees.size.positive?

    CaseContactsContactDates.new(interviewees).contact_dates_details
  end
```

All this could really test is that the `ActiveRecord` query is correct, and its not a very complex query. Is there value in it?

Reply:
I am somewhat unclear on what an interviewee is
In general I am in favor of more testing rather than less tho